config/routes.rb

```diff
@@ -4,6 +4,10 @@
   root to: 'pages#home'
   
   scope "/:locale" do
-    resources :projects
-  end
+    resources :pages do
```

Comment:
This is a bit unconventional. If you'd prefer, we can merge it as-is and create an issue to adjust things in a separate PR.

Typically, our controllers will somewhat mirror our models. That is, if we plan on having a Rails model "project" (which it seems like we would, based on your form), then we'd have a `projects_controller.rb` (The C in the "MVC" pattern). There would be a route for `/projects/new`, which we'd define with `resources :projects, only: [:new]`. This will allow all project-related actions (view, create, update) to be handled within the projects controller with routes like `/projects/:id`, `/projects/:id/edit`, etc. Rails defines most of that by convention for us. 

It's worth noting at this point that rails provides [generators](https://guides.rubyonrails.org/command_line.html#bin-rails-generate) to do things like scaffold a controller for us.

See https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default for info about how routing is typically set up.

Reply:
I figured there would be some rails generators for stuff like this, but being my first hands-on rails project, I have no idea how the generators work yet. I'm not sure what to expect when running a scaffolding generator, and I still have to understand better what controllers are. So I'm certain this can be done in a better / different way, I just don't have the experience to do it yet. It's an accomplishment for me to have succeeded in getting the button on the home page to go to this form page! So I would say, let's go ahead and merge this, then create an issue and fix it up in a separate PR, because I'm sure someone else will know how to do it better than me right now.

Reply:
in fact I created a "How tos" section in the wiki to document stuff like this. It would be great to add to that wiki page a practical example of how a generator was used to create a Rails model "project".

Reply:
this is related to the discussion in issue #5 , so I'd say that after this has been pulled, we can try to break down the sections of the UI to see how they relate to the scaffolding and the user experience workflow. I added a couple more comments on the above issue in these regards.

Reply:
I just tested using the scaffolding generator on a separate branch, to see what it does and what it looks like.

I ran: 
```ruby
bundle exec rails generate scaffold Project liturgy:integer brideFirstName:string brideLastName:string groomFirstName:string groomLastName:string celebrantNamePrefix:string celebrantFirstName:string celebrantLastName:string church:string city:string weddingdate:datetime
```

Then I ran:
```ruby
bundle exec rails db:migrate RAILS_ENV=development
```

Now when I run the webserver, I can see the basic unstyled form which is automatically created at `localhost:3000/en/projects/new`, and I see the links to `create`, `edit`, `update`, `back` in the different views. I see how it allows to create a new project which will then be listed at `localhost:3000/en/projects/`, and will be accessible at `localhost:3000/en/projects/1`.

Is there any way to use a unique hash instead of a progressive integer, to identify resources that are created?

Reply:
It seems this is the way to use a hash: https://pawelurbanek.com/uuid-order-rails